### PR TITLE
EZP-29539 : Deleting object will remove all subtree items even when user does not have permission to delete them

### PR DESCRIPTION
**Pull request by pawbuj**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-29539](https://jira.ez.no/browse/EZP-29539)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

returns to the same page on error

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Controller/LocationController.php [hunk 1]
```diff
@@ -354,7 +354,7 @@ public function trashAction(Request $request): Response
             }
         }
 
-        return $this->redirect($this->generateUrl('ezplatform.trash.list'));
+        return $this->redirectToLocation($form->getData()->getLocation());
```
Comment (alongosz) [inline src/bundle/Controller/LocationController.php]: why do you need to change behavior?

**Comment by mikadamczyk** [comment]
Should it be merged to 1.2? Also in the newest version is the same bug but in `trashContatinerAction`

File: src/bundle/Controller/LocationController.php (hunk 1 above)
Comment (pawbuj) [reply src/bundle/Controller/LocationController.php]: The user tried to move the location to the trash but the operation failed so he should stay on the same page and see the error message. Another issue is that we do not know if the user has permissions to see the trash list.

Comment (adamwojs) [reply src/bundle/Controller/LocationController.php]: IMHO this fragment should looks as follow: 

```php
if (($form->getData()->getLocation()) !== null)
    return $this->redirectToLocation($location);
}

return $this->redirectToRoute('ezplatform.trash.list');
```

Even if reaching `$this->redirectToLocation($location);` instruction is not possible from UI, the request shouldn't end-up with "500: Internal Server Error".

Comment (pawbuj) [reply src/bundle/Controller/LocationController.php]: @adamwojs fixed

**Comment by lserwatka** [comment]
Could you merge it up?

**Comment by pawbuj** [comment]
merged

**Pull request merged**